cmd/tailscale/cli/cli.go

```diff
@@ -216,11 +238,29 @@ var rootArgs struct {
 	socket string
 }
 
-func walkCommands(cmd *ffcli.Command, f func(*ffcli.Command)) {
-	f(cmd)
+func walkCommands(cmd *ffcli.Command, parents []*ffcli.Command, f func(c *ffcli.Command, parents []*ffcli.Command)) {
```

Comment:
the old signature was readable. this new signature probably warrants some docs on what's going on.

Reply:
maybe make the recursive part be a closure func inside this func and keep the outer func exposed to callers not taking the parents second argument that is always nil?

And maybe the func can take a new struct type containing the command (embedded?) and parents, rather than taking two things, especially because the second thing is often ignored by callers?

optional

Reply:
I've rewritten this function to use iteration because it was upsetting me that I couldn't reference the recursive closure from its own declaration: https://go.dev/play/p/JZcwMW7ttD0

Reply:
The way to write self-referential closures is https://go.dev/play/p/18WqmO49FTF for better or worse